shared/services/wallet/wallet.go

```diff
@@ -75,12 +114,38 @@ type walletStore struct {
 }
 
 // Create new wallet
-func NewWallet(walletPath string, chainId uint, maxFee *big.Int, maxPriorityFee *big.Int, gasLimit uint64, passwordManager *passwords.PasswordManager) (*Wallet, error) {
+func NewWallet(addressPath string, walletPath string, chainId uint, maxFee *big.Int, maxPriorityFee *big.Int, gasLimit uint64, passwordManager *passwords.PasswordManager, addressManager *AddressManager) (Wallet, error) {
+
+	// Check if the address file exists
+	_, err := os.Stat(addressPath)
+	masquerading := !os.IsNotExist(err)
```

Comment:
I'd move this under the conditional at L123-5, it's always good to put error handling right after the line that returns an error.